src/LowerTriangularArrays/lower_triangular_array.jl

```diff
@@ -526,31 +579,31 @@ function LowerTriangularMatrix{T}(M::LowerTriangularMatrix{T2}) where {T,T2}
 end
 
 function Base.convert(
-    ::Type{LowerTriangularArray{T1, N, ArrayTypeT1}},
-    L::LowerTriangularArray{T2, N, ArrayTypeT2},
-) where {T1, T2, N, ArrayTypeT1<:AbstractArray{T1}, ArrayTypeT2<:AbstractArray{T2}}
-    return LowerTriangularArray{T1, N, ArrayTypeT1}(L.data, L.m, L.n)
+    ::Type{LowerTriangularArray{T1, N, ArrayTypeT1, S}},
+    L::LowerTriangularArray{T2, N, ArrayTypeT2, S},
+) where {T1, T2, N, S <: AbstractSpectrum, ArrayTypeT1<:AbstractArray{T1}, ArrayTypeT2<:AbstractArray{T2}}
+    return LowerTriangularArray{T1, N, ArrayTypeT1, S}(L.data, L.spectrum)
 end
 
 function Base.convert(::Type{LowerTriangularMatrix{T}}, L::LowerTriangularMatrix) where T
-    return LowerTriangularMatrix{T}(L.data, L.m, L.n)
+    return LowerTriangularMatrix{T,typeof(L.spectrum)}(L.data, L.spectrum)
 end
 
-function Base.similar(::LowerTriangularArray{T, N, ArrayType}, I::Integer...) where {T, N, ArrayType}
-    return LowerTriangularArray{T,N,ArrayType}(undef, I...)
+function Base.similar(::LowerTriangularArray{T, N, ArrayType, SP}, I::Integer...) where {T, N, ArrayType, SP}
```

Comment:
I believe `similar` methods should share the `spectrum` of the similar lowertriangular array?

Reply:
You are absolutely right in principle, I also adjusted this for the third similar constructor now. But it's a bit unclear to me how handle this with the `I` here. So far you were able to allocate a similar LTA with different dimensions with this routine. In these cases you couldn't share/reuse the `spectrum`. We could also remove that possibility and just interpret `I` as the trailing batch dimensions. What do you think?

Reply:
e.g. I wrote this
```julia
Base.similar(field::AbstractField) = Field(similar(field.data), field.grid)
```

Reply:
Ok, I've done that now. I also added some tests that test that's really the same spectrum.